src/include/platform/ConnectivityManager.h

```diff
@@ -60,8 +60,25 @@ struct NetworkInterface : public app::Clusters::GeneralDiagnostics::Structs::Net
     NetworkInterface * Next; /* Pointer to the next structure.  */
 };
 
+class ConnectivityManager;
 class ConnectivityManagerImpl;
 
+/**
+ * Defines the delegate class of Connectivity Manager to notify connectivity updates.
+ */
+class ConnectivityManagerDelegate
+{
+public:
+    virtual ~ConnectivityManagerDelegate() {}
+
+    /**
+     * @brief
+     *   Called when any network interface on the Node is changed
+     *
+     */
+    virtual void OnNetworkInfoChanged() {}
```

Comment:
Uh... Nothing is ever calling this, so it doesn't work in practice afaict.

Reply:
Yes, as in the PR description, it will be done in future since I have not figured out a good mechanisms to monitor the network interface change event on Linux reference platform yet. One possible why is to listen to the event from wpa_supplicant via D-Bus, but it only contain the wireless interface.